src/core/or/circuit_st.h

```diff
@@ -177,6 +182,25 @@ struct circuit_t {
   /** Hashtable node: used to look up the circuit by its HS token using the HS
       circuitmap. */
   HT_ENTRY(circuit_t) hs_circuitmap_node;
+
+  /** Adaptive Padding state machines: these are immutable. The state machines
+   *  that come from the consensus are saved to a global structure, to avoid
+   *  per-circuit allocations. This merely points to the global copy.
+   *
+   *  Each element of this array corresponds to a different padding machine,
+   *  and we can have up to CIRCPAD_MAX_MACHINES such machines. */
+  const circpad_machine_t *padding_machine[CIRCPAD_MAX_MACHINES];
```

Comment:
We need to make sure that the global structure itself never changes somewhere, and is never allocated or deallocated, right?

Reply:
Maybe add comments here and in the code that sets up the foo_padding_machines?

Reply:
Done in 94c33eabc7.